loki/transformations/array_indexing.py

```diff
@@ -46,21 +46,40 @@ def remove_explicit_array_dimensions(routine, calls_only=False):
     ----------
     routine: :any:`Subroutine`
         The subroutine to check
+    calls_only: bool
+        Whether to remove colon notation from array dimensions only
+        from arrays within (inline) calls or all arrays (default: False)
     """
     if calls_only:
+        # handle calls (to subroutines) and inline calls (to functions)
         calls = FindNodes(ir.CallStatement).visit(routine.body)
-        for call in calls:
+        inline_calls = FindInlineCalls().visit(routine.body)
+        inline_call_map = {}
+        for call in as_tuple(calls) + as_tuple(inline_calls):
+            # handle arguments
             arguments = ()
             for arg in call.arguments:
-                if isinstance(arg, sym.Array):
-                    if all(dim == sym.RangeIndex((None, None)) for dim in arg.dimensions):
-                        new_dimensions = None
-                        arguments += (arg.clone(dimensions=new_dimensions),)
-                    else:
-                        arguments += (arg,)
+                if isinstance(arg, sym.Array) and all(dim == sym.RangeIndex((None, None)) for dim in arg.dimensions):
+                    new_dimensions = None
+                    arguments += (arg.clone(dimensions=new_dimensions),)
                 else:
                     arguments += (arg,)
-            call._update(arguments=arguments)
+            # handle kwargs
+            kwarguments = ()
+            for (kwarg_name, kwarg) in call.kwarguments:
+                if isinstance(kwarg, sym.Array) and all(dim==sym.RangeIndex((None, None)) for dim in kwarg.dimensions):
+                    kwarguments += ((kwarg_name, kwarg.clone(dimensions=None)),)
+                else:
+                    kwarguments += ((kwarg_name, kwarg),)
```

Comment:
This line is currently untested. For full coverage, you could modify the `test_transform_excplicit_dimensions` to use a call like this (incorporating @mlange05's suggestion):
```
call kernel_routine(nlon, a, b=b, nlev=nlev)
```